Tests `test_ldap_external_user_directory` occasionally flaky

Integration tests `test_ldap_external_user_directory` are [flaky](https://play.clickhouse.com/play?user=play#c2VsZWN0IHB1bGxfcmVxdWVzdF9udW1iZXIgYXMgcHIsIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdGVzdF9zdGF0dXMgYXMgc3RhdHVzLCByZXBvcnRfdXJsCmZyb20gY2hlY2tzCndoZXJlIHRvRGF0ZShjaGVja19zdGFydF90aW1lKSA+ICcyMDIzLTA3LTI4JyBhbmQgdGVzdF9uYW1lIGxpa2UgJyV0ZXN0X2xkYXBfZXh0ZXJuYWxfdXNlcl9kaXJlY3RvcnklJyBhbmQgc3RhdHVzIGluICgnRkFJTCcsICdFUlJPUicsICdGTEFLWScpCm9yZGVyIGJ5IGNoZWNrX3N0YXJ0X3RpbWUgZGVzYw==)

Example [report](https://s3.amazonaws.com/clickhouse-test-reports/63206/bf93c3807014f4323960afb953565a48e3c26109/integration_tests__aarch64__[4_6].html)

It is interesting that it has not failed for months, and now it only fails in aarch64.

```
=================================== FAILURES ===================================
___________________________ test_authentication_pass ___________________________
[gw3] linux -- Python 3.10.12 /usr/bin/python3

    def test_authentication_pass():
>       assert instance.query(
            "select currentUser()", user="janedoe", password="qwerty"
        ) == TSV([["janedoe"]])

test_ldap_external_user_directory/test.py:52: 

.........................................................................
.........................................................................

E           helpers.client.QueryRuntimeException: Client failed! Return code: 4, stderr: Code: 516. DB::Exception: Received from 172.16.9.3:9000. DB::Exception: janedoe: Authentication failed: password is incorrect, or there is no user with such name.. Stack trace:
E           
E           0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c343e08
E           1. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x0000000007b9f21c
E           2. DB::AccessControl::authenticate(DB::Credentials const&, Poco::Net::IPAddress const&, String const&) const @ 0x000000000f10cce4
E           3. DB::Session::authenticate(DB::Credentials const&, Poco::Net::SocketAddress const&) @ 0x000000001024ff20
E           4. DB::TCPHandler::receiveHello() @ 0x00000000112643b4
E           5. DB::TCPHandler::runImpl() @ 0x000000001125ab4c
E           6. DB::TCPHandler::run() @ 0x0000000011273048
E           7. Poco::Net::TCPServerConnection::start() @ 0x00000000135ae698
E           8. Poco::Net::TCPServerDispatcher::run() @ 0x00000000135aeb74
E           9. Poco::PooledThread::run() @ 0x0000000013579574
E           10. Poco::ThreadImpl::runnableEntry(void*) @ 0x0000000013577950
E           11. ? @ 0x000000000007d5c8
E           12. ? @ 0x00000000000e5edc
E           . (AUTHENTICATION_FAILED)

helpers/client.py:239: QueryRuntimeException
------------------------------ Captured log call -------------------------------
2024-06-26 04:06:01 [ 517 ] DEBUG : Executing query select currentUser() on instance (cluster.py:3449, query)
```

--- a/tests/integration/compose/docker_compose_ldap.yml
+++ b/tests/integration/compose/docker_compose_ldap.yml
@@ -15,7 +15,10 @@ services:
         ports:
             - ${LDAP_EXTERNAL_PORT:-1389}:${LDAP_INTERNAL_PORT:-1389}
         healthcheck:
-            test: "ldapsearch -x -b dc=example,dc=org cn > /dev/null"
+            test: >
+                ldapsearch -x -H ldap://localhost:$$LDAP_PORT_NUMBER -D $$LDAP_ADMIN_DN -w $$LDAP_ADMIN_PASSWORD -b $$LDAP_ROOT
+                | grep -c -E "member: cn=j(ohn|ane)doe"
+                | grep 2 >> /dev/null
             interval: 10s
             retries: 10
             timeout: 2s

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -2640,7 +2640,9 @@ class ClickHouseCluster:
                     [
                         "bash",
                         "-c",
-                        f"/opt/bitnami/openldap/bin/ldapsearch -x -H ldap://{self.ldap_host}:{self.ldap_port} -D cn=admin,dc=example,dc=org -w clickhouse -b dc=example,dc=org",
+                        f'/opt/bitnami/openldap/bin/ldapsearch -x -H ldap://{self.ldap_host}:{self.ldap_port} -D cn=admin,dc=example,dc=org -w clickhouse -b dc=example,dc=org'
+                        f'| grep -c -E "member: cn=j(ohn|ane)doe"'
+                        f'| grep 2 >> /dev/null',
                     ],
                     user="root",
                 )

--- a/tests/integration/helpers/cluster.py
+++ b/tests/integration/helpers/cluster.py
@@ -2640,9 +2640,9 @@ class ClickHouseCluster:
                     [
                         "bash",
                         "-c",
-                        f'/opt/bitnami/openldap/bin/ldapsearch -x -H ldap://{self.ldap_host}:{self.ldap_port} -D cn=admin,dc=example,dc=org -w clickhouse -b dc=example,dc=org'
+                        f"/opt/bitnami/openldap/bin/ldapsearch -x -H ldap://{self.ldap_host}:{self.ldap_port} -D cn=admin,dc=example,dc=org -w clickhouse -b dc=example,dc=org"
                         f'| grep -c -E "member: cn=j(ohn|ane)doe"'
-                        f'| grep 2 >> /dev/null',
+                        f"| grep 2 >> /dev/null",
                     ],
                     user="root",
                 )

